glean-core/android/src/main/java/mozilla/telemetry/glean/net/PingUploader.kt

```diff
@@ -15,22 +17,34 @@ typealias HeadersList = List<Pair<String, String>>
  *
  * See below for the different possible cases.
  */
-sealed class UploadResult
+sealed class UploadResult {
+    open fun toFfi(): Int {
+        return Constants.UPLOAD_RESULT_UNRECOVERABLE
+    }
+}
 
 /**
  * A HTTP response code.
  *
  * This can still indicate an error, depending on the status code.
  */
-data class HttpResponse(val statusCode: Int) : UploadResult()
+data class HttpResponse(val statusCode: Int) : UploadResult() {
+    override fun toFfi(): Int {
+        return Constants.UPLOAD_RESULT_HTTP_STATUS or statusCode
```

Comment:
Having come from a C background, the fact that Kotlin uses `and` and `or` rather than `&` and `|` is kinda weird.  It's definitely more readable but I have a tendency to look at the words as the logical varieties and not the bitwise ones.

Reply:
I wrote this first with `|` just for the compiler to shout at me.
I was pleasantly surprised that it uses words.